components/srs_page/Flashcard.tsx

```diff
@@ -0,0 +1,104 @@
+import React, { useState, useEffect } from 'react';
+import { View, Text, TouchableOpacity } from 'react-native';
+import { Card } from '@/lib/superMemov2';
+import { Audio } from 'expo-av';
+import { Volume2 } from 'lucide-react-native';
+
+interface FlashcardProps {
+  card: Card;
+  onGrade: (card: Card, grade: number) => void;
+}
+
+const Flashcard: React.FC<FlashcardProps> = ({ card, onGrade }) => {
+  const [revealed, setRevealed] = useState(false);
+  const [sound, setSound] = useState<Audio.Sound | null>(null);
+
+  useEffect(() => {
+    setRevealed(false);
+    return () => {
+      if (sound) {
+        sound.unloadAsync();
+      }
+    };
+  }, [card]);
+
+  const playAudio = async (audioUrl: string) => {
+    if (sound) {
+      await sound.stopAsync();
+      sound.unloadAsync();
+    }
+
+    const { sound: newSound } = await Audio.Sound.createAsync({ uri: audioUrl });
+    setSound(newSound);
+    await newSound.playAsync();
+  };
+
+  return (
+    <View className="flex-1 w-full max-w-md justify-center items-center p-4">
+      <View className="bg-blue-100 p-6 rounded-lg shadow-md w-full">
+      <View className="flex flex-row justify-between items-center relative">
+          <View className="flex-1">
+            <Text className="text-3xl font-bold text-gray-800 text-center">{card.englishWord}</Text>
+          </View>
+          <TouchableOpacity onPress={() => playAudio(card.audioSource)} className="absolute right-0">
+            <Volume2 className="h-5 w-5 text-gray" />
+          </TouchableOpacity>
+        </View>
+        {revealed ? (
+          <Text className="text-2xl mt-4 text-center text-gray-700">{card.japaneseWord}</Text>
+        ) : (
+          <TouchableOpacity
+            onPress={() => setRevealed(true)}
+            className="mt-4 bg-blue-500 px-6 py-2 rounded-full"
+          >
+            <Text className="text-lg text-white text-center">Reveal</Text>
+          </TouchableOpacity>
+        )}
+      </View>
+
+      {revealed && (
+        <View className="mt-6 w-full">
+          {/* First row: Something and No Clue */}
+          <View className="flex flex-row justify-between mb-4">
+            <TouchableOpacity
+              className="flex-1 border-2 border-red-500 px-4 py-2 mx-1 rounded-lg bg-white"
+              onPress={() => onGrade(card, 1)}
```

Comment:
I see how this works now. 

Small point but `onGrade` sounds like something that should happen after the grade action, e.g. think 'onClick'. This is actually grading it though isn't it? 

Maybe this is a better name

```suggestion
              onPress={() => gradeItem(card, 1)}
```

Reply:
Personally I think 5 options is too many from a UX perspective - perhaps 3 would be better so as not to overwhelm the user?

Reply:
> I see how this works now.
> 
> Small point but `onGrade` sounds like something that should happen after the grade action, e.g. think 'onClick'. This is actually grading it though isn't it?
> 
> Maybe this is a better name

Fair point, maybe onGradeClick would be more understandable?

Reply:
I think the point is it's not happening 'on' something. This function is being called as a callback from 'onPress' so we should name it better so it actually describes what it does. The action is grading isn't it? In that case just 'grade' or 'gradeItem' seems best.

Reply:
naming things well is the hardest part of programming btw - this is not a joke!

Reply:
Ah right I see what you mean now. I've seen "handle" used in what I think are similar cases. Would it be misleading to use handleGrade or handleGradeItem? (Just curious!)

Reply:
I forgot the function was defined as handleGrade in the parent component, I'll pass it as gradeItem as you suggested to the Flashcard component.